pkg/client/cache/delta_fifo.go

```diff
@@ -43,13 +45,13 @@ import (
 //       affects error retrying.
 //
 // Also see the comment on DeltaFIFO.
-func NewDeltaFIFO(keyFunc KeyFunc, compressor DeltaCompressor, knownObjectKeys KeyLister) *DeltaFIFO {
+func NewDeltaFIFO(keyFunc KeyFunc, compressor DeltaCompressor, knownObjects KeyLookup) *DeltaFIFO {
```

Comment:
Erm, i guess you mean a simple type assertion. Sure. 
I still think it should be upto the clients, but doing as suggested should be easy enough.